api/agent_test.go

```diff
@@ -155,6 +155,16 @@ func TestAPI_AgentMembersOpts(t *testing.T) {
 	if len(members) != 2 {
 		t.Fatalf("bad: %v", members)
 	}
+
+	members, err = agent.MembersOpts(MembersOpts{
+		WAN:    true,
+		Filter: `Tags["dc"] == dc2`,
+	})
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	require.Equal(t, 1, len(members))
```

Comment:
could we test the nil result case here too? 
i.e. Tags["dc"] == doesNotExist

Reply:
@skpratt , that's a good point. More test cases are added. Thanks.